CoreDNS is burning CPU and spamming my network with DNS requests to 1.1.1.1

Sorry I don't know what logs to capture to help pinpoint the issue more precisely, so please let me know if you'd like me to capture appropriate diagnostics.

I noticed that my Home Assistant OS running in a VM on an Intel NUC was using a lot more CPU than I remembered.  And after logging in via SSH, I saw that the `coredns` process is using anywhere from 40-60% of the virtual CPU and had burned over 900 hours of processing time since its last restart!

As it happens, my router has been having connectivity issues lately so on a hunch I ran a tcpdump.  There I found a constant endless stream of DNS requests to Cloudflare's 1.1.1.1 DNS service originating from the server that hosts my Home Assistant instance.

```
15:53:55.887610 IP XXX.lan.50080 > one.one.one.one.853: Flags [S], seq 4108605629, win 64240, options [mss 1460,sackOK,TS val 1419242137 ecr 0,nop,wscale 7], length 0
15:53:55.887936 IP one.one.one.one.853 > XXX.lan.50080: Flags [R.], seq 0, ack 4108605630, win 0, length 0
```

My router is configured to hijack DNS requests from the local network for privacy reasons.  When I disable that feature, the spam stops and the excess CPU usage vanishes.  So Home Assistant seems not to like the responses that it is receiving and it just keeps trying again and again, flooding the network.

Why is Home Assistant attempting to bypass my locally configured DNS server?  And why is it retrying so aggressively?

Home Assistant Version: core-2022.5.0
Supervisor Version: supervisor-2022.05.0

Just did a first installation of Home Assistant OS on rpi3 just to give it a try and test some of zigbee-enabled hardware. My pi is burning with 70+ degrees with `coredns` in the top for CPU time and I'm seeing this in logs:

```
➜  log docker logs hassio_dns 2>/dev/null | tail -30
[INFO] 127.0.0.1:38539 - 28343 "NS IN . udp 17 false 512" NOERROR - 0 5.007165535s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.1.1.1:853: connect: connection refused
[INFO] 127.0.0.1:58223 - 23638 "NS IN . udp 17 false 512" NOERROR - 0 5.001772123s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.0.0.1:853: connect: connection refused
[INFO] 127.0.0.1:43302 - 63044 "NS IN . udp 17 false 512" NOERROR - 0 5.001283552s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.1.1.1:853: connect: connection refused
[INFO] 127.0.0.1:51113 - 35362 "NS IN . udp 17 false 512" NOERROR - 0 5.003702062s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.0.0.1:853: connect: connection refused
[INFO] 127.0.0.1:38262 - 56755 "NS IN . udp 17 false 512" NOERROR - 0 5.002333335s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.0.0.1:853: connect: connection refused
[INFO] 127.0.0.1:33174 - 9323 "NS IN . udp 17 false 512" NOERROR - 0 5.003397732s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.0.0.1:853: connect: connection refused
[INFO] 127.0.0.1:33094 - 32776 "NS IN . udp 17 false 512" NOERROR - 0 5.010711032s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.0.0.1:853: connect: connection refused
[INFO] 127.0.0.1:44416 - 48101 "NS IN . udp 17 false 512" NOERROR - 0 5.002663142s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.1.1.1:853: connect: connection refused
[INFO] 127.0.0.1:35474 - 16537 "NS IN . udp 17 false 512" NOERROR - 0 5.005863059s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.1.1.1:853: connect: connection refused
[INFO] 127.0.0.1:50759 - 43058 "NS IN . udp 17 false 512" NOERROR - 0 5.000412148s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.0.0.1:853: connect: connection refused
[INFO] 127.0.0.1:33519 - 45040 "NS IN . udp 17 false 512" NOERROR - 0 5.004900032s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.0.0.1:853: connect: connection refused
[INFO] 127.0.0.1:39739 - 59567 "NS IN . udp 17 false 512" NOERROR - 0 5.005401149s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.0.0.1:853: connect: connection refused
[INFO] 127.0.0.1:39157 - 50430 "NS IN . udp 17 false 512" NOERROR - 0 5.00624565s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.0.0.1:853: connect: connection refused
[INFO] 127.0.0.1:48527 - 47596 "NS IN . udp 17 false 512" NOERROR - 0 5.003011352s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.0.0.1:853: connect: connection refused
[INFO] 127.0.0.1:60637 - 59663 "NS IN . udp 17 false 512" NOERROR - 0 5.001902262s
[ERROR] plugin/errors: 2 . NS: dial tcp 1.1.1.1:853: connect: connection refused
```

So my guess is that I'm hitting one of the issues highlighted in this report. I do have an openwrt-powered router with DoH enabled. Any pointers on how to normalize this are greatly appreciated.

Version info: Home Assistant 2023.10.3 Supervisor 2023.10.0 Operating System 11.0 Frontend 20231005.0 - latest

--- a/rootfs/etc/corefile
+++ b/rootfs/etc/corefile
@@ -10,22 +10,34 @@
     template ANY AAAA local.hass.io hassio {
         rcode NOERROR
     }
+    template ANY A local.hass.io hassio {
+        rcode NXDOMAIN
+    }
     mdns
-    forward . dns://172.0.0.11:53 dns://127.0.0.1:5553 {
+    forward . dns://127.0.0.11 {
         except local.hass.io
         policy sequential
         health_check 1m
+        max_fails 5
     }
+    fallback REFUSED,SERVFAIL,NXDOMAIN . dns://127.0.0.1:5553
     cache
 }
 
 .:5553 {
     log
     errors
+    template IN NS . {
+        rcode REFUSED
+    }
+    template IN HINFO . {
+        rcode REFUSED
+    }
     forward . tls://1.1.1.1 tls://1.0.0.1 {
         tls_servername cloudflare-dns.com
+        max_fails 2
         except local.hass.io
-        health_check 5m
+        health_check 10m
     }
-    cache
+    cache 600
 }

--- a/rootfs/usr/share/tempio/corefile
+++ b/rootfs/usr/share/tempio/corefile
@@ -30,11 +30,18 @@
         class error
     }{{ end }}
     errors
+    template IN NS . {
+        rcode REFUSED
+    }
+    template IN HINFO . {
+        rcode REFUSED
+    }
     {{ if .debug }}debug{{ end }}
     forward . tls://1.1.1.1 tls://1.0.0.1 {
         tls_servername cloudflare-dns.com
-        max_fails 0
+        max_fails 2
         except local.hass.io
+        health_check 10m
     }
     cache 600
 }

